File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int low=Integer.MAX_VALUE,high=0;
        for(int i:prices){
            if(i<low){
                low=i;
            }
            if((i-low)>high)high=i-low;
        }
        return high;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        // int low=prices[0],res=0;
        // for(int i=1;i<prices.size();i++){
        //     if(prices[i]<low){
        //         low=prices[i];
        //     }
        //     else res=max(prices[i]-low,res);
        // }
        // return res;
        
        int low=prices[0];
        int diff=0;
        for(int i=1;i<prices.size();i++){
            if(prices[i]<low)low=prices[i];
            else diff=max(diff,prices[i]-low);
        }
        return diff;
        
    }
};